### ISSUE TITLE
ttp text template parser produce 

### ISSUE THREAD

**Issue by scottpeterman** [Nuitka/Nuitka#1820]
ttp is a text template parser. it compiles but reports a logic error when you run it... This is very easy to reproduce. just download the ttp project and run: 

(venv) C:\Users\----\ttp_tool\ttp-master\ttp-master>nuitka --follow-imports ttp\ttp.py
Nuitka-Options:INFO: Used command line options: --follow-imports ttp\ttp.py
Nuitka:INFO: Starting Python compilation with Nuitka '1.1' on Python '3.8' commercial grade 'not installed'.
Nuitka-Plugins:INFO: multiprocessing: Injecting pre-module load code for module 'multiprocessing':
Nuitka-Plugins:INFO: multiprocessing:     Monkey patching "multiprocessing" load environment.
Nuitka-Plugins:INFO: multiprocessing: Injecting post-module load code for module 'multiprocessing':
Nuitka-Plugins:INFO: multiprocessing:     Monkey patching "multiprocessing" for compiled methods.
Nuitka:INFO: Completed Python level compilation and optimization.
Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: gcc (gcc).
Nuitka-Scons:INFO: Backend linking program with 12 files (no progress information available).
Nuitka-Scons:INFO: Compiled 24 C files using ccache.
Nuitka-Scons:INFO: Cached C files (using ccache) with result 'cache hit': 12
Nuitka:INFO: Keeping build directory 'ttp.build'.
Nuitka:INFO: Successfully created 'ttp.exe'.
Nuitka:INFO: Execute it by launching 'ttp.cmd', the batch file needs to set environment.

(venv) C:\Users\97685\ttp_tool\ttp-master\ttp-master>ttp.cmd
Traceback (most recent call last):
  File "C:\Users\97685\ttp_tool\ttp-master\ttp-master\ttp\ttp.py", line 3667, in <module>
    cli_tool()
  File "C:\Users\97685\ttp_tool\ttp-master\ttp-master\ttp\ttp.py", line 3637, in cli_tool
    parser_Obj.add_template(template=ttp_template)
UnboundLocalError: local variable 'ttp_template' referenced before assignment

ttp seems to have circular references (lazy loading ?) 

ttp project: https://github.com/dmulyalin/ttp

from ttp.py:
    # create parser object
    parser_Obj = ttp(base_path=BASE_PATH, vars=ttp_vars)  <---- ttp self referencing class ttp

# load templates file
    if TEMPLATE_FILE:
        ttp_template = _ttp_["utils"]["load_files"](TEMPLATE_FILE, read=True)[0][1]
        if TEMPLATE_NAME:
            ttp_template = _ttp_["utils"]["load_struct"](
                text_data=ttp_template, load="python"
            )
            ttp_template = ttp_template[TEMPLATE_NAME]
    # add data and templates
    parser_Obj.add_template(template=ttp_template)  <--- Line causing error
    if DATA:
        parser_Obj.set_input(data=DATA)
        timing("Data descriptors loaded")
    timing("Template(s) loaded")

**Comment by kayhayen** [comment]
In the template, I have made it clear, that not filling out the issue template is not acceptable. I am closing this kind of report as invalid. The main reason is that I do not want to be left guessing. For you it's easy to erase the issue template and expect me to ask you questions only as necessary, but I cannot do that. I would much rather wait for you to create a new report, fill out the template, or wait for somebody with the necessary skill to do it.

In short: No issue template, no fun.
